pkg/controller/clusterinstallation/utils.go

```diff
@@ -112,14 +112,21 @@ func (r *ReconcileClusterInstallation) checkClusterInstallation(mattermost *matt
 	return status, nil
 }
 
-func (r *ReconcileClusterInstallation) checkSecret(secretName, namespace string) error {
+func (r *ReconcileClusterInstallation) checkSecret(secretName, keyName, namespace string) (bool, error) {
 	foundSecret := &corev1.Secret{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: namespace}, foundSecret)
 	if err != nil {
-		return errors.Wrap(err, "Error getting secret")
+		return false, errors.Wrap(err, "Error getting secret")
 	}
 
-	return nil
+	for key := range foundSecret.Data {
```

Comment:
This should probably be:
```suggestion
	for _, key := range foundSecret.Data {
```